src/process/bvoc/CCPr_BVOC_Mod.F90

```diff
@@ -0,0 +1,389 @@
+
+      ! INPUT/OUTPUT PARAMETERS
+      !------------------------
+      TYPE(BvocStateType), intent(inout) :: BvocState ! Bvoc state
+      INTEGER,              intent(inout) :: RC         ! Success or failure
+
+      ! Error handling
+      !---------------
+      CHARACTER(LEN=255)    :: ErrMsg
+      CHARACTER(LEN=255)    :: ThisLoc
+
+      ! LOCAL VARIABLES
+      !----------------
+      INTEGER               :: c
+
+      ! Put any local variables here
+
+      !=================================================================
+      ! CCPR_BVOC_Init begins here!
+      !=================================================================
+      RC = CC_SUCCESS
+      ThisLoc = ' -> at CCPR_BVOC_INIT (in process/bvoc/ccpr_bvoc_mod.F90)'
+
+      ! First check if process is activated in config | if not don't allocate arrays or pointers
+      if (Config%Bvoc_activate) then
+
+         ! Activate Process
+         !------------------
+         BvocState%Activate = .true.
+
+         ! Set Scheme Options
+         !-------------------
+         if (Config%bvoc_scheme < 0) then ! not listed in config
+            BvocState%SchemeOpt = 1
+         else
+            BvocState%SchemeOpt = Config%bvoc_scheme
+         endif
+
+         ! CO2 inhibition option
+         !TODO: what if it is not given in the configuration file properly
+         !------------------
+         BvocState%CO2Inhib = Config%megan_co2_inhib
+
+         ! Set CO2 concentration (ppm)
+         ! In case it is given as a negative value in config
+         !----------------------------
+         if (Config%megan_co2_conc_ppm < 0) then
+            BvocState%CO2conc = 390.0_fp
+         else
+            BvocState%CO2conc = Config%megan_co2_conc_ppm
+         endif
+
+         ! Check GLOBCO2 if CO2 inhibition is turned on (LISOPCO2 = .TRUE.)
+         ! GLOBCO2 should be between 150-1250 ppmv. Isoprene response to
+         ! CO2 outside this range has no empirical basis.
+         if ( BvocState%CO2Inhib ) then
+            if ( BvocState%CO2conc <  150.0_fp .or. &
+               BvocState%CO2conc > 1250.0_fp     ) then
+               RC = CC_FAILURE
+               ErrMsg = 'Global CO2 outside valid range of 150-1250 ppmv!'
+               call CC_Error( errMsg, RC, thisLoc )
+               return
+            endif
+         endif
+
+         !--------------------------------------------
+         !Find bvoc caterory index in EmisState for future use
+         do c = 1, EmisState%nCats
+            if (EmisState%Cats(c)%name == 'BVOC') then
+               BvocState%CatIndex = c
+               exit
+            endif
+         end do
+
+         ! Set number of species from EmisState
+         !----------------------
+         BvocState%nBvocSpecies = EmisState%Cats(BvocState%CatIndex)%nSpecies
+
+         !------------------------------------
+         ! Allocate emission species index
+         ALLOCATE( BvocState%BvocSpeciesIndex(BvocState%nBvocSpecies) )
```

Comment:
```suggestion
         ALLOCATE( BvocState%BvocSpeciesIndex(BvocState%nBvocSpecies), STAT=RC )
```
(and similarly for the others below) otherwise the `CC_CheckVar` isn't doing anything